Lib/fontTools/otlLib/optimize/gpos.py

```diff
@@ -0,0 +1,405 @@
+import logging
+from collections import defaultdict
+from functools import reduce
+from itertools import chain
+from math import log2
+from typing import DefaultDict, Dict, Iterable, List, Sequence, Tuple
+
+from fontTools.misc.intTools import bit_count, bit_indices
+from fontTools.ttLib import TTFont
+from fontTools.ttLib.tables import otBase, otTables
+
+GPOS_COMPACT_MODE_ENV_KEY = "FONTTOOLS_GPOS_COMPACT_MODE"
+GPOS_COMPACT_MODE_DEFAULT = "0"
+
+log = logging.getLogger("fontTools.otlLib.optimize.gpos")
+
+
+def compact(font: TTFont, mode: str) -> TTFont:
+    # Plan:
+    #  1. Find lookups of Lookup Type 2: Pair Adjustment Positioning Subtable
+    #     https://docs.microsoft.com/en-us/typography/opentype/spec/gpos#lookup-type-2-pair-adjustment-positioning-subtable
+    #  2. Extract glyph-glyph kerning and class-kerning from all present subtables
+    #  3. Regroup into different subtable arrangements
+    #  4. Put back into the lookup
```

Comment:
Maybe this needs to be clarified. 
In compact_ext_lookup() when it calls compact_pair_pos(), each subtable is treated independently as it runs compact_class_pairs() for each subtable.
For the description, one would expect the extension lookup subtables to be treated and reorganized together instead of individually. Saying `3. Split subtables into more smaller subtables` would be clearer.

Reply:
Done, I clarified that the plan was the ideal plan and that the actual implementation was as you pointed out